xarray/core/duck_array_ops.py

```diff
@@ -715,12 +709,16 @@ def mean(array, axis=None, skipna=None, **kwargs):
     if dtypes.is_datetime_like(array.dtype):
         offset = _datetime_nanmin(array)
 
-        # xarray always uses np.datetime64[ns] for np.datetime64 data
-        dtype = "timedelta64[ns]"
+        # From version 2025.01.2 xarray uses np.datetime64[unit], where unit
+        # is one of "s", "ms", "us", "ns".
+        # To not have to worry about the resolution, we just convert the output
+        # to "int64" and let the dtype of offset take precedence.
+        # This is fully backwards compatible with datetime64[ns]. For other
+        # this might lead to imprecise output, where fractional parts are truncated.
```

Comment:
```suggestion
        # From version 2025.01.2 xarray uses np.datetime64[unit], where unit
        # is one of "s", "ms", "us", "ns".
        # To not have to worry about the resolution, we just convert the output
        # to "timedelta64" (without unit) and let the dtype of offset take precedence.
        # This is fully backwards compatible with datetime64[ns]. For other
        # this might lead to imprecise output, where fractional parts are truncated.
```